src/main/java/taxi/dao/impl/DriverDaoJdbcImpl.java

```diff
@@ -0,0 +1,110 @@
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.DriverDao;
+import taxi.lib.Dao;
+import taxi.model.Driver;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class DriverDaoJdbcImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        String sqlQuery = "INSERT INTO drivers (driver_name, driver_lisence_number)"
+                + "VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(sqlQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, driver.getName());
+            preparedStatement.setString(2, driver.getLicenceNumber());
+            preparedStatement.executeUpdate();
+            ResultSet resultset = preparedStatement.getGeneratedKeys();
+            while (resultset.next()) {
+                driver.setId(resultset.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert driver " + driver, e);
+        }
+        return driver;
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String sqlQuery = "SELECT * FROM drivers WHERE drivers_id =?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery);) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultset = preparedStatement.executeQuery();
+            Driver newDriver = null;
+            if (resultset.next()) {
+                newDriver = createDriver(resultset);
+            }
+            return Optional.ofNullable(newDriver);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get driver by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String query = "SELECT * FROM drivers WHERE driver_deleted =false;";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultset = preparedStatement.executeQuery();
+            while (resultset.next()) {
+                drivers.add(createDriver(resultset));
+            }
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't getAll drivers ", ex);
+        }
+        return drivers;
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String query = "UPDATE drivers SET driver_name=?, driver_lisence_number=?"
+                + " WHERE drivers_id =? AND driver_deleted = 'FALSE';";
```

Comment:
```suggestion
                + " WHERE drivers_id =? AND driver_deleted = FALSE;";
```

Reply:
In case change i receive error in this method "Can't update driver Driver / Data truncation: Truncated incorrect DOUBLE value: 'FALSE'". Maybe it due type of my driver_deleted. It's VARCHAR(5)! I set it folow checklist recomendation where was suggested use true/false unstead 0/1.

Reply:
use  TINYINT() for bollean values